cmd/controller/main.go

```diff
@@ -65,6 +65,7 @@ func main() {
 
 	config := controllerruntime.GetConfigOrDie()
 	config.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(float32(opts.KubeClientQPS), opts.KubeClientBurst)
+	config.UserAgent = "karpenter"
```

Comment:
thoughts on including the version as well? 

```
config.UserAgent = fmt.Sprintf("karpenter-%s", project.Version)
```

Reply:
If karpenter-A sets some field and then version change to B, karpenter-B will have a conflict with itself.

See https://kubernetes.io/docs/reference/using-api/server-side-apply/

Reply:
o wow, I did not realize server side apply used user-agent... https://kubernetes.io/docs/reference/using-api/server-side-apply/#:~:text=For%20other%20updates%2C%20its%20default%20is%20computed%20from%20the%20user%2Dagent.